src/app/admin/pages/edit-book/edit-book.component.ts

```diff
@@ -0,0 +1,73 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup } from '@angular/forms';
+import { ActivatedRoute, Router } from '@angular/router';
+
+import { BookCategories } from 'src/app/book/constants/books';
+import { BookModel } from 'src/app/core/models/book';
+import { HttpDataService } from 'src/app/core/services/http-data/http-data.service';
+
+@Component({
+  selector: 'app-edit-book',
+  templateUrl: './edit-book.component.html',
+  styleUrls: ['./edit-book.component.scss'],
+})
+export class EditBookComponent implements OnInit {
+  checkoutForm = new FormGroup({});
+  @Input() book!: BookModel;
+  categories = BookCategories;
+  constructor(
+    private route: ActivatedRoute,
+    private httpDataService: HttpDataService,
+    private formBuilder: FormBuilder,
+    public router: Router,
+  ) { }
+
+  ngOnInit(): void {
+    this.book = this.getBook();
+    this.checkoutForm = this.formBuilder.group({
+      bookName: this.book.name,
+      bookDescription: this.book.description,
+      bookPrice: this.book.price,
+      bookCategory: this.book.category,
+      bookDate: this.book.createDate,
+      available: this.book.isAvailable,
+    });
+  }
+
+  getBook(): BookModel {
+    return  this.route.snapshot.data.book || {
+      name: '',
+      description: '',
+      price: 0,
+      category: BookCategories.Romance,
+      createDate: 0,
+      isAvailable: false,
+    };
+  }
+
+  updateBook() {
+    this.book.name = this.checkoutForm.value.bookName;
```

Comment:
we discussed it on the prev call and nothing has changes, why?

Reply:
fixed